Keas.Mvc/Handlers/VerifyAuthTokenHandler.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Http;
+using Keas.Mvc.Attributes;
+using Keas.Mvc.Services;
+using Keas.Core.Domain;
+using System.Linq;
+
+namespace Keas.Mvc.Handlers
+{
+    public class VerifyRoleOrAuthTokenHandler : AuthorizationHandler<VerifyRoleOrAuthToken> {
+        private readonly ISecurityService _securityService;
+        private readonly ITempDataDictionaryFactory _tempDataDictionaryFactory;
+        private readonly IHttpContextAccessor _httpContext;
+
+        public VerifyRoleOrAuthTokenHandler(ISecurityService securityService, IHttpContextAccessor httpContext, ITempDataDictionaryFactory tempDataDictionary)
+        {
+            _securityService = securityService;
+            _httpContext = httpContext;
+            _tempDataDictionaryFactory = tempDataDictionary;
+        }
+
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, VerifyRoleOrAuthToken requirement) {
+            // get team name from url
+            var team = "";
+            if (context.Resource is AuthorizationFilterContext mvcContext)
+            {
+                if (mvcContext.RouteData.Values["teamName"] != null)
+                {
+                    team = mvcContext.RouteData.Values["teamName"].ToString();
+                }
+            }
+            if (string.IsNullOrWhiteSpace(team))
+            {
+                var tempData = _tempDataDictionaryFactory.GetTempData(_httpContext.HttpContext);
+                team = Convert.ToString(tempData["TeamName"]);
+            }
+
+            // user has access to team based on claims
+
+            if (context.User.HasClaim("APIKEYTEAM", team)) {
+                context.Succeed(requirement);
+            }
```

Comment:
Is this supposed to fall through, or should there be a return?

Reply:
@sprucely good catch -- I think the fallthrough is functionally fine, but it would be better to just quick return once we get success.  I'll change it